onnxruntime/core/providers/cuda/cuda_execution_provider.cc

```diff
@@ -211,12 +211,23 @@ void CUDAExecutionProvider::PerThreadContext::IncrementRegularRunCountBeforeGrap
 #endif
 
 void OverrideTunableOpInfoByEnv(CUDAExecutionProviderInfo& info) {
-  auto env_tunable_op_enabled = onnxruntime::ParseTestOnlyEnvironmentVariable<bool>(
-      "ORT_CUDA_TUNABLE_OP_ENABLED", {"0", "1"}, "Use provider_options \"tunable_op_enabled\" instead.");
-  if (env_tunable_op_enabled.has_value() && env_tunable_op_enabled != info.tunable_op.enabled) {
+  if (auto env_tunable_op_enabled = onnxruntime::ParseTestOnlyEnvironmentVariable<bool>(
+          "ORT_CUDA_TUNABLE_OP_ENABLED", {"0", "1"}, "Use provider_options \"tunable_op_enabled\" instead.");
+      env_tunable_op_enabled.has_value() && env_tunable_op_enabled != info.tunable_op.enabled) {
     LOGS_DEFAULT(INFO) << "ORT_CUDA_TUNABLE_OP_ENABLED is set to " << *env_tunable_op_enabled;
     info.tunable_op.enabled = *env_tunable_op_enabled;
   }
+
+  if (auto env_tunable_op_tuning = onnxruntime::ParseTestOnlyEnvironmentVariable<bool>(
+          "ORT_CUDA_TUNABLE_OP_TUNING", {"0", "1"}, "Use provider_options \"tunable_op_tuning\" instead.");
+      env_tunable_op_tuning.has_value() && env_tunable_op_tuning != info.tunable_op.tuning) {
+    LOGS_DEFAULT(INFO) << "ORT_CUDA_TUNABLE_OP_TUNING is set to " << *env_tunable_op_tuning;
+    info.tunable_op.tuning = *env_tunable_op_tuning;
+  }
+
+  if (info.tunable_op.tuning && !info.tunable_op.enabled) {
+    LOGS_DEFAULT(WARNING) << "TunableOp is enabled for tuning but is not enabled for using. This will have no effect.";
```

Comment:
> "TunableOp is enabled for tuning but is not enabled for using. This will have no effect."

If you renable the flags `enable_tunable_op` and `enable_op_tuning` as suggested in another comment, then you can replace this warning with something like: "Op tuning has been enabled without enabling tunable ops. This will have no effect"

Or perhaps it is cleaner to just have `enable_op_tuning ==> enable_tunable_ops`. Thoughts?